mlflow/gateway/providers/openai.py

```diff
@@ -122,27 +110,25 @@ async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
         # }
         # ```
         return chat.ResponsePayload(
-            **{
-                "candidates": [
-                    {
-                        "message": {
-                            "role": c["message"]["role"],
-                            "content": c["message"]["content"],
-                        },
-                        "metadata": {
-                            "finish_reason": c["finish_reason"],
-                        },
-                    }
-                    for c in resp["choices"]
-                ],
-                "metadata": {
-                    "input_tokens": resp["usage"]["prompt_tokens"],
-                    "output_tokens": resp["usage"]["completion_tokens"],
-                    "total_tokens": resp["usage"]["total_tokens"],
-                    "model": resp["model"],
-                    "route_type": self.config.route_type,
-                },
-            }
+            id=resp["id"],
+            object=resp["object"],
+            created=resp["created"],
+            model=resp["model"],
+            choices=[
+                chat.Choice(
+                    index=idx,
+                    message=chat.ResponseMessage(
+                        role=c["message"]["role"], content=c["message"]["content"]
+                    ),
+                    finish_reason=c["finish_reason"],
+                )
+                for idx, c in enumerate(resp["choices"])
+            ],
+            usage=chat.ChatUsage(
```

Comment:
curious what would happen if we did 
```python
usage = chat.ChatUsage(resp["usage"])
```
or is this a guard in case they add a new field?

Reply:
Yeah, this is a guard in case of new fields